pkg/controller/virtualmachineimport/virtualmachineimport_controller.go

```diff
@@ -343,39 +399,308 @@ func (r *ReconcileVirtualMachineImport) addWatchForImportPod(instance *v2vv1.Vir
+	if len(jobList.Items) > 0 {
+		return &jobList.Items[0], nil
+	}
+	return nil, nil
+}
+
+func (r *ReconcileVirtualMachineImport) makeGuestConversionJobSpec(vmSpec *kubevirtv1.VirtualMachine) *batchv1.Job {
+	// Only ever run the guest conversion job once per VM
+	completions := int32(1)
+	parallelism := int32(1)
+	backoffLimit := int32(0)
+
+	volumes, volumeMounts := makeJobVolumeMounts(vmSpec)
+
+	return &batchv1.Job{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Job",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: virtV2vJobName + "-",
+			Namespace:    vmSpec.Namespace,
+			Labels: map[string]string{
+				VMLabel: vmSpec.Name,
+			},
+		},
+		Spec: batchv1.JobSpec{
+			Completions:  &completions,
+			Parallelism:  &parallelism,
+			BackoffLimit: &backoffLimit,
+			Template: v1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					GenerateName: virtV2vJobName + "-",
+					Namespace:    vmSpec.Namespace,
+				},
+				Spec: v1.PodSpec{
+					RestartPolicy: v1.RestartPolicyNever,
+					Containers: []v1.Container{
+						{
+							Name:            virtV2vJobName,
+							Image:           virtV2vImage,
+							ImagePullPolicy: v1.PullIfNotPresent,
+							VolumeMounts:    volumeMounts,
+						},
+					},
+					Volumes: volumes,
+				},
+			},
+		},
+		Status: batchv1.JobStatus{},
+	}
+}
+
+func makeJobVolumeMounts(vmSpec *kubevirtv1.VirtualMachine) ([]v1.Volume, []v1.VolumeMount) {
+	volumes := make([]v1.Volume, 0)
+	volumeMounts := make([]v1.VolumeMount, 0)
+	// add volumes and mounts for each of the VM's disks.
+	// the virt-v2v pod expects to see the disks mounted at /mnt/disks/diskX
+	for i, dataVolume := range vmSpec.Spec.Template.Spec.Volumes {
+		vol := v1.Volume{
+			Name: dataVolume.DataVolume.Name,
+			VolumeSource: v1.VolumeSource{
+				PersistentVolumeClaim: &v1.PersistentVolumeClaimVolumeSource{
+					ClaimName: dataVolume.DataVolume.Name,
+					ReadOnly:  false,
+				},
+			},
+		}
+		volumes = append(volumes, vol)
+
+		volMount := v1.VolumeMount{
+			Name:      dataVolume.DataVolume.Name,
+			MountPath: fmt.Sprintf("/mnt/disks/disk%v", i),
+		}
+		volumeMounts = append(volumeMounts, volMount)
+	}
+
+	// add volume and mount for the libvirt domain xml config map.
+	// the virt-v2v pod expects to see the libvirt xml at /mnt/v2v/input.xml
+	volumes = append(volumes, v1.Volume{
+		Name: vmSpec.Name,
+		VolumeSource: v1.VolumeSource{
+			ConfigMap: &v1.ConfigMapVolumeSource{
+				LocalObjectReference: v1.LocalObjectReference{
+					Name: vmSpec.Name,
+				},
+			},
+		},
+	})
+	volumeMounts = append(volumeMounts, v1.VolumeMount{
+		Name:      vmSpec.Name,
+		MountPath: "/mnt/v2v",
+	})
+	return volumes, volumeMounts
+}
+
+// findLibvirtDomainConfigMap finds the libvirt domain xml configmap for a given VM, returning nil if it can't be found.
+func (r *ReconcileVirtualMachineImport) findLibvirtDomainConfigMap(vmName types.NamespacedName) (*v1.ConfigMap, error) {
+	configMapList := &v1.ConfigMapList{}
```

Comment:
What do you think about using https://github.com/kubevirt/vm-import-operator/blob/master/pkg/configmaps/configmaps.go?

Reply:
Makes sense. It might also make sense to create a similar package for the Jobs so they follow the same pattern. What do you think?

Reply:
Yes, good idea.